nailgun/entities.py

```diff
@@ -2031,6 +2031,17 @@ def __init__(self, server_config=None, **kwargs):
         }
         super(ContentCredential, self).__init__(server_config, **kwargs)
 
+    def update_payload(self, fields=None):
+        """Override the method to only update fields that actually can be updated
+        """
+        updatable_fields = set.intersection({'name', 'content_type', 'content'},
```

Comment:
Any reason why you're doing the intersection here, instead of just setting these fields directly? Concern about a field getting removed from the entity?

Reply:
Because I hit the issue with not all fields being set on initialization, leading to errors when calling the update_payload. However, I don't know *why* they weren't set.

Reply:
Would be good to include that as an inline comment, so anybody else reading/using this method understands why its done this way.